ui/packages/shared/components/src/Callgraph/index.tsx

```diff
@@ -34,13 +33,13 @@ const Callgraph = ({graph, sampleUnit, width}: Props): JSX.Element => {
       setGraphData(jsonGraph);
     };
 
-    if (width) {
-      getDataWithPositions();
+    if (width !== null) {
+      void getDataWithPositions();
     }
-  }, [width]);
+  }, [graph, width]);
 
   // 3. Render the graph with calculated layout in Canvas container
-  if (!width || !graphData) return <></>;
+  if (width == null || graphData == null) return <></>;
 
   const height = width;
   const {objects, edges: gvizEdges, bb: boundingBox} = JSON.parse(graphData);
```

Comment:
`edge` can be typed earlier. I think this is the earliest it can be (with these other variables). Then TypeScript should be able to figure out the types of the derived variables from `edges` being `IEdge[]`